hydro_lang/src/live_collections/stream/mod.rs

```diff
@@ -781,32 +847,28 @@ where
         self.assume_retries::<AtLeastOnce>(nondet)
     }
 
-    /// Weakens the retries guarantee provided by the stream to be the weaker of the
-    /// current guarantee and `R2`. This is safe because the output guarantee will
-    /// always be weaker than the input.
-    pub fn weaken_retries<R2>(self) -> Stream<T, L, B, O, <R as MinRetries<R2>>::Min>
-    where
-        R: MinRetries<R2>,
-    {
+    /// Weakens the retries guarantee provided by the stream to `R2`, with the type-system
+    /// enforcing that `R2` is weaker than the input retries guarantee.
+    pub fn weaken_retries<R2: Retries + MinRetries<R, Min = R2>>(self) -> Stream<T, L, B, O, R2> {
```

Comment:
Seems like the old API may be more usable? In that you can put in whatever and let it compute the min. Although to force the user to think about what they're getting out

Also `MinRetires` implies `Retries`